c/parallel/src/reduce.cu

```diff
@@ -283,38 +228,54 @@ CUresult cccl_device_reduce_build(
 
     const std::string op_src = make_kernel_user_binary_operator(accum_cpp, accum_cpp, accum_cpp, op);
 
+    const std::string ptx_arch = std::format("-arch=compute_{}{}", cc_major, cc_minor);
+
+    constexpr size_t ptx_num_args      = 5;
+    const char* ptx_args[ptx_num_args] = {ptx_arch.c_str(), cub_path, thrust_path, libcudacxx_path, "-rdc=true"};
+
     const std::string src = std::format(
-      R"XXX(
-#include <cub/block/block_reduce.cuh>
-#include <cub/device/dispatch/kernels/reduce.cuh>
-struct __align__({1}) storage_t {{
-  char data[{0}];
-}};
-{4}
-{5}
-struct agent_policy_t {{
-  static constexpr int ITEMS_PER_THREAD = {2};
-  static constexpr int BLOCK_THREADS = {3};
-  static constexpr int VECTOR_LOAD_LENGTH = {7};
-  static constexpr cub::BlockReduceAlgorithm BLOCK_ALGORITHM = cub::BLOCK_REDUCE_WARP_REDUCTIONS;
-  static constexpr cub::CacheLoadModifier LOAD_MODIFIER = cub::LOAD_LDG;
-}};
-struct device_reduce_policy {{
-  struct ActivePolicy {{
-    using ReducePolicy = agent_policy_t;
-    using SingleTilePolicy = agent_policy_t;
-  }};
-}};
-{6}
-)XXX",
+      "#include <cub/block/block_reduce.cuh>\n"
+      "#include <cub/device/dispatch/kernels/reduce.cuh>\n"
+      "struct __align__({1}) storage_t {{\n"
+      "  char data[{0}];\n"
+      "}};\n"
+      "{2}\n"
+      "{3}\n"
+      "{4}\n",
       input_it.value_type.size, // 0
       input_it.value_type.alignment, // 1
-      policy.items_per_thread, // 2
-      policy.block_size, // 3
-      input_iterator_src, // 4
-      output_iterator_src, // 5
-      op_src, // 6
-      policy.vector_load_length); // 7
+      input_iterator_src, // 2
+      output_iterator_src, // 3
+      op_src); // 4
+
+    nlohmann::json runtime_policy = get_policy(
+      std::format("cub::detail::reduce::MakeReducePolicyWrapper(cub::detail::reduce::policy_hub<{}, {}, "
+                  "{}>::MaxPolicy::ActivePolicy{{}})",
+                  accum_cpp,
+                  offset_t,
+                  "op_wrapper"),
+      "#include <cub/device/dispatch/tuning/tuning_reduce.cuh>\n" + src,
+      ptx_args);
+
+    using cub::detail::RuntimeReduceAgentPolicy;
+    auto [reduce_policy, reduce_policy_str] = RuntimeReduceAgentPolicy::from_json(runtime_policy, "ReducePolicy");
+    auto [st_policy, st_policy_str]         = RuntimeReduceAgentPolicy::from_json(runtime_policy, "SingleTilePolicy");
+    auto [segment_policy,
+          segment_policy_str] = RuntimeReduceAgentPolicy::from_json(runtime_policy, "SegmentedReducePolicy");
```

Comment:
I don't think we need `segment_policy` here because we have a separate `segmented_reduce`